lib/elixir/src/elixir_expand.erl

```diff
@@ -921,6 +922,26 @@ assert_generator_start(Meta, _, E) ->
 
 %% Assertions
 
+refute_parallel_bitstring_match({'<<>>', _, _}, {'<<>>', Meta, _} = Arg, E, true) ->
+  form_error(Meta, ?key(E, file), ?MODULE, {parallel_bitstring_match, Arg});
+refute_parallel_bitstring_match(Expr, {'=', _Meta, [Left, Right]}, E, Parallel) ->
+  refute_parallel_bitstring_match(Expr, Left, E, true),
+  refute_parallel_bitstring_match(Expr, Right, E, Parallel);
+refute_parallel_bitstring_match(Expr, Arg, E, Parallel) when is_list(Expr), is_list(Arg) ->
+  refute_parallel_bitstring_match_each(Expr, Arg, E, Parallel);
+refute_parallel_bitstring_match(Expr, Arg, E, Parallel) when tuple_size(Expr) == 2, tuple_size(Arg) == 2 ->
```

Comment:
this is a bit uncommon in the Elixir codebase. Can we pattern match on the tuple and build the lists explicitly?

    refute_parallel_bitstring_match({Expr1, Expr2}, {Arg1, Arg2}, E, Parallel) ->
       ...

Reply:
Yeah, I went with `tuple_size/1` to not deal with naming: it is not really `Expr1`, `Expr2`, `Arg1`, `Arg2` here.

Reply:
`{Left1, Left2}, {Right1, Right2}` is used on parallel traversals from time to time. But I think in avoiding the naming issue we are making the code less clear so we should find a name that you are at least comfortable with.

Reply:
Agree, `{Left1, Left2}, {Right1, Right2}` is exactly what I switched to.